components/scream/cmake/ScreamUtils.cmake

```diff
@@ -16,16 +16,31 @@ FUNCTION(CreateUnitTest target_name target_srcs scream_libs)
   cmake_parse_arguments(CreateUnitTest "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
 
   # Set link directories (must be done BEFORE add_executable is called)
-  link_directories(${SCREAM_TPL_LIBRARY_DIRS} ${SCREAM_LIBRARY_DIRS})
+  link_directories(${SCREAM_TPL_LIBRARY_DIRS} ${SCREAM_LIBRARY_DIRS} ${UNIT_TEST_EXTRA_LIBRARY_DIRS})
 
   # Create the executable
   add_executable (${target_name} ${target_srcs} ${SCREAM_SRC_DIR}/share/util/scream_catch_main.cpp)
 
+  SET (INCLUDE_DIRS
```

Comment:
I would prefer that you use cmake_parse_arguments to get these additional optional arguments, like was done for CONFIG_DEFS.

Reply:
Sounds good. On this topic, I noticed that `SCREAM_LIBRARY_DIRS` only contains `${CMAKE_SOURCE_DIR}/src`. Our libs do not get built there though, right? Should we remove this variable? Should we populate it with something else? I think it is not needed though, since cmake should already automatically recognize libs built within the project, and grab the correct directory for those, no?

Reply:
Actually, if what I said is true, I do not even need the `UNIT_TEST_EXTRA_LIBRARY_DIRS` variable...

Reply:
Yup, just checked, and things work without those two variables. I'm going to push a commit removing them.

Reply:
Nice.